qemu/tests/qemu_guest_agent.py

```diff
@@ -343,6 +343,24 @@ def gagent_verify(self, params, vm):
         self.gagent.verify_responsive()
         logging.info(self.gagent.cmd("guest-info"))
 
+    @error_context.context_aware
+    def gagent_setsebool_verify(self, params, vm):
+        session = self._get_session(params, self.vm)
+        self._open_session_list.append(session)
+        error_context.context("Check selinux boolean value.", logging.info)
+        set_selinux_bool_cmd = self.params['setsebool_cmd']
+        get_selinux_bool_cmd = self.params['getsebool_cmd']
+        set_selinux_bool_guest = session.cmd(set_selinux_bool_cmd).strip()
+        value_selinux_bool_guest = session.cmd_output(get_selinux_bool_cmd).strip()
+        if value_selinux_bool_guest != 'on':
+            self.test.error("Set selinux boolean virt_qemu_ga_read_nonsecurity_files failed.\n"
+                            "Selinux boolean virt_qemu_ga_read_nonsecurity_files is not %s."
+                            % value_selinux_bool_guest)
+        else:
+            logging.info("Selinux boolean virt_qemu_ga_read_nonsecurity_files is %s."
+                         % value_selinux_bool_guest)
+        session.close()
```

Comment:
if you use self._open_session_list.append(session),there is no need to close.
￼

Reply:
And why don't you use the session already existed.